src/main/java/com/google/cloud/tools/maven/StageMojo.java

```diff
@@ -216,15 +215,15 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
     getLog().info("Staging the application to: " + stagingDirectory);
 
-    appengineWebXml =
-        new AppEngineWebXml(new File(sourceDirectory.toString() + "/WEB-INF/appengine-web.xml"));
-
     configureAppEngineDirectory();
 
     if (isStandardStaging()) {
       getLog().info("Detected App Engine standard environment application.");
 
       // force runtime to 'java' for compat projects using Java version >1.7
+      AppEngineWebXml appengineWebXml =
+          new AppEngineWebXml(
+              new File(sourceDirectory.toPath().resolve("WEB-INF/appengine-web.xml").toString()));
```

Comment:
`.resolve("WEB-INF").resolve("appengine-web.xml")`

but also, maybe I missed some older comments, why did we need to move this around?

Reply:
I wanted to call `isStandardStaging()` in `DeployAllMojo`, but the original implementation used `appengineWebXml`, which was just a private field that was set in `StageMojo.execute(). So whenever I tried to call `isStandardStaging()`, `appengineWebXml` was null and I'd get an error. So I changed `isStandardStaging()` to not rely on the field, realized `appengineWebXml` wasn't used anywhere else, and reduced its scope.